File: dart/lib/src/bridge_exception.dart
import 'protocol.dart';

/// Exception type for bridge protocol errors. The Dart equivalent of .NET's
/// `BridgeException` / Swift's `BridgeError`.
class BridgeException implements Exception {
  /// One of the [BridgeErrorType] values.
  final String errorType;

  /// Human-readable error message.
  final String message;

  BridgeException(this.errorType, [String? message])
      : message = message ?? errorType;

  @override
  String toString() => 'BridgeException($errorType): $message';
}

File: android/aspectly-bridge/src/main/kotlin/com/aspectly/bridge/BridgeException.kt
package com.aspectly.bridge

import com.aspectly.bridge.protocol.BridgeErrorType

/**
 * Exception type for bridge protocol errors. The Kotlin equivalent of .NET's
 * `BridgeException` / Swift's `BridgeError`.
 */
class BridgeException(
    val errorType: BridgeErrorType,
    message: String? = null,
    cause: Throwable? = null,
) : Exception(message ?: errorType.value, cause)
